plugins/inputs/p4runtime/p4runtime.go

```diff
@@ -0,0 +1,232 @@
+package p4runtime
+
+import (
+	"context"
+	"crypto/tls"
+	_ "embed"
+	"fmt"
+	"io"
+	"sync"
+
+	p4ConfigV1 "github.com/p4lang/p4runtime/go/p4/config/v1"
+	p4v1 "github.com/p4lang/p4runtime/go/p4/v1"
+	"golang.org/x/exp/slices"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/influxdata/telegraf"
+	internaltls "github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+const (
+	defaultDeviceID = 1
+	defaultEndpoint = "127.0.0.1:9559"
+)
+
+type P4runtime struct {
+	Endpoint            string          `toml:"endpoint"`
+	DeviceID            uint64          `toml:"device_id"`
+	CounterNamesInclude []string        `toml:"counter_names_include"`
+	Log                 telegraf.Logger `toml:"-"`
+	EnableTLS           bool            `toml:"enable_tls"`
+	internaltls.ClientConfig
+
+	conn   *grpc.ClientConn
+	client p4v1.P4RuntimeClient
+	wg     sync.WaitGroup
+}
+
+func (*P4runtime) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *P4runtime) Init() error {
+	if p.Endpoint == "" {
+		p.Log.Debugf("Using default Endpoint: %v", defaultEndpoint)
+		p.Endpoint = defaultEndpoint
+	}
+
+	return p.newP4RuntimeClient()
+}
+
+func (p *P4runtime) Gather(acc telegraf.Accumulator) error {
+	p4Info, err := p.getP4Info()
+	if err != nil {
+		return err
+	}
+
+	if len(p4Info.Counters) == 0 {
+		p.Log.Warn("No counters available in P4 Program!")
+		return nil
+	}
+
+	filteredCounters := filterCounters(p4Info.Counters, p.CounterNamesInclude)
+	if len(filteredCounters) == 0 {
+		p.Log.Warn("No filtered counters available in P4 Program!")
+		return nil
+	}
+
+	p.wg.Add(len(filteredCounters))
+
+	for _, counter := range filteredCounters {
```

Comment:
I think this is better as it will also work if we break/exit the for loop and later try to `Stop()` the plugin...
```suggestion
	for _, counter := range filteredCounters {
		p.wg.Add(1)

```

Reply:
applied